Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Migration/Tools/MigrationWindow.cs

```diff
@@ -0,0 +1,417 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using UnityEngine.SceneManagement;
+using UnityEditor.SceneManagement;
+
+using Object = UnityEngine.Object;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Utilities
+{
+    /// <summary>
+    /// This tool allows the migration of game object obsolete components into up-to-date versions.
+    /// Requires Implementation of IMigrationhandler.
+    /// </summary>
+    public class MigrationWindow : EditorWindow
```

Comment:
>MigrationWindow [](start = 17, length = 15)

Do folks have any thoughts on if this window would also handle non-game object/component upgrade scenarios? i.e. I'm thinking about situations where we make asmdef changes (again, god help us), or make changes to fields of components which might not be... migratable, I'm not sure if such a situation exists.

Not a blocking thought for this review, just wanted to hear thoughts on this question.

Reply:
hmmm now that you're mentioning it - we might need that for other asset files as well - like scriptable objects that are not components / directly associated to gameobjects

Reply:
though it's probably alright to add that functionality once we have something to upgrade in those areas - i don't want to invest too much time now for something we might not use in the end :)